C_CPP/Fundamentals/Operators & Expressions/cast_operators.md

```diff
@@ -0,0 +1,161 @@
+    <li>Constant Cast</li>
+    <li>Dynamic Cast</li>
+    <li>Reinterpret Cast</li>
+</ul>
+<p> Let's look into each of them breifly!</p>
+
+
+<h1 id="1">Static Cast</h1>
+
+- This is used for any standard conversion of data types. 
+- It could even convert pointers of derived class to base class and vice versa.
+- It helps in converting int,char,float,double pointers to void pointer and vice versa.
+
+**Syntax**
+
+> *static_cast &lt;data_type&gt;(var_name);*
+
+<p> Now,lets see some examples of how it is used! </p>
+
+```
+
+#include<iostream>
+#include<stdlib.h>
+
+using namespace std;
+
+int main(){
+    int a = 97;                             //int variable 
+    float b = static_cast<float>(a);        //conversion of int to float
+    char c = static_cast<char>(a);          //conversion of int to char
+    double d = static_cast<double>(c);      //conversion of char to double
+    int* iptr = &a;                         //pointer to int type variable
+    void* vptr = static_cast<void*>(iptr);  //conversion of int* to void*
+    int* intptr = static_cast<int*>(vptr);  //conversion of void* to int*
+    return 0;
+}
+```
+
+
+<h1 id="2">Constant Cast</h1>
+
+- This is helpful in removing constantness of a variable. 
+
+**Syntax**
+> *const_cast&lt;data_type&gt;(var_name);*
+
+- Now, lets see how it is used!
+
+```
+#include<iostream>
+#include<stdlib.h>
+
+using namespace std;
+
+int main(){
+    float a = 52.61;
+    const float* const_ptr = &a;
+    //float* f = const_ptr;                    This gives an error!
+    float* f = const_cast<float*>(const_ptr);  
+    return 0;
+}
+```
+
+<h1 id="3">Dynamic Cast</h1>
+
+- It is especially used for converting pointers to objects.
+- It does the conversion at **run time**.
+- It could safely convert pointers and references to classes up, down, and sideways along the inheritance hierarchy. 
+- Downcasting is only possible if base class is **polymorphic**
+
+**Syntax**
+
+> *dynamic_cast < data_type > ( var_name )*
+
+- It tries to convert *var_name* to a type of *data_type* and if conversion is successful, it returns pointer of type *data_type*. Else, it would simply return a null pointer.
+
+- It could always convert a pointer of derived class to base class.
+- It could also convert a pointer of base class to derived.
+
+- Let's see how this is done with an example.
+
+```
+#include<iostream>
+using namespace std;
+
+class Base { virtual void vf(){} };
+
+class Derived : public Base { };
+
+int main() 
+{
+	Base *pBDerived = new Derived;
+	Base *pBBase = new Base;
+	Derived *pd;
+
+	pd = dynamic_cast<Derived*>(pBDerived);	//1
+    if(pd != nullptr)
+        cout << "Successfully converted pBDerived to base class pointer";
+        
+	pd = dynamic_cast<Derived*>(pBBase);      //2 
+    if(pd != nullptr)
+        cout << "Successfully converted pBBase to base class pointer";
+
+	return 0;
+}
+```
+The output for the above code is:
+> *Successfully converted pBDerived to base class pointer*

```

Comment:
what do you mean by pBDerived?

Reply:
pointer of Base class type pointing a derived class

Reply:
Should i add that as a comment? @ChetasShree

Reply:
Yes , add it as a comment

Reply:
Ok, I'll add the comments. Are there any other changes to be made?

Reply:
Can you write pointer of Base class type pointing a derived class instead of pBDerived

Reply:
Ok